execution/src/execution.cc

```diff
@@ -261,15 +277,30 @@ namespace leatherman { namespace execution {
 
         out_stream.open(out_file.c_str(), std::ios::binary);
         if (!out_stream.is_open()) {
-            throw execution_exception(_("failed to open the output file."));
+            throw execution_exception(_("failed to open output file {1}", out_file));
+        }
+
+        boost::system::error_code ec;
+        if (perms) {
+            fs::permissions(out_file, *perms, ec);
+            if (ec) {
+                throw execution_exception(_("failed to modify permissions on output file {1} to {2,num,oct}: {3}", out_file, *perms, ec.message()));
+            }
         }
```

Comment:
I wish there was a way in C++ to open a file and set its permissions in a single step (e.g. like in C with the [`open`](http://man7.org/linux/man-pages/man2/open.2.html) function). As if done in two steps like here, there is still a short window when someone can open the file before its permissions are set more restrictively.